cmd/access/node_builder/staked_access_node_builder.go

```diff
@@ -86,23 +91,61 @@ func (builder *StakedAccessNodeBuilder) Initialize() cmd.NodeBuilder {
 }
 
 func (anb *StakedAccessNodeBuilder) Build() AccessNodeBuilder {
-	anb.FlowAccessNodeBuilder.
-		Build().
-		Component("ping engine", func(builder cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
-			ping, err := pingeng.New(
-				node.Logger,
-				node.State,
-				node.Me,
-				anb.PingMetrics,
-				anb.pingEnabled,
-				node.Middleware,
-				anb.nodeInfoFile,
-			)
-			if err != nil {
-				return nil, fmt.Errorf("could not create ping engine: %w", err)
-			}
-			return ping, nil
-		})
+	anb.FlowAccessNodeBuilder.Build()
+
+	if anb.supportsUnstakedFollower {
+		var unstakedNetworkConduit network.Conduit
+		var proxyEngine *splitter.Engine
+
+		anb.
+			Component("unstaked sync request proxy", func(builder cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
```

Comment:
if unstakedNetworkConduit and proxyEngine is being used only in the `unstaked sync request handler` component, why create it here as a separate component and not combine it one single component that sets up the sync request handler for the staked AN.

Reply:
only because `proxyEngine` is `ReadyDoneAware` and we need to call `Ready` on it. No other reason really